### DISCUSSION TITLE
Coveralls Github Action

### DISCUSSION THREAD

**Discussion by helio-frota** [nodeshift/new-module-starter#12] [General]
I think we have some others repositories using local coveralls instead of coveralls github action.

It would be good for us to follow this approach https://github.com/nodeshift/npcheck/pull/55

**Comment by richardlau** [comment]
Yep, we currently have three repositories in this org using the Coveralls module: https://github.com/search?q=org%3Anodeshift+coveralls.js&type=code
I can probably make a start converting those tomorrow.

**Reply by richardlau** [reply]
- [x] [opossum-prometheus](https://github.com/nodeshift/opossum-prometheus/blob/82f33158cc9b81a142cc056d594012f4873f21d9/.github/workflows/nodejs-ci-action.yml) https://github.com/nodeshift/opossum-prometheus/pull/53
- [x] [license-reporter](https://github.com/nodeshift/license-reporter/blob/ec742a89a55e98b98ea936e57d6d37bd92eba013/.github/workflows/nodejs-ci-action.yml) https://github.com/nodeshift/license-reporter/pull/389
- [x] [oratrix](https://github.com/nodeshift/oratrix/blob/aa9e40bd04385d428391aaad770561b0a393865a/.github/workflows/nodejs-ci-action.yml) https://github.com/nodeshift/oratrix/pull/99

**Comment by helio-frota** [comment]
- [x] nodejs-rest-http https://github.com/nodeshift-starters/nodejs-rest-http/pull/210
- [x] https://github.com/nodeshift-starters/nodejs-health-check/pull/224
- [x] https://github.com/nodeshift-starters/nodejs-configmap/pull/195
- [x] https://github.com/nodeshift-starters/nodejs-rest-http-crud/pull/243

**Comment by richardlau** [comment]
FYI I came across https://github.com/lemurheavy/coveralls-public/issues/1531 which is why I think we have coverage failures using the action -- there's a bug on Coveralls' end where if no thresholds are set no change (i.e. 0%) is being flagged as failing.

**Reply by helio-frota** [reply]
so probably we need to set that in coveralls for each repo right?

**Reply by richardlau** [reply]
I think so. Did we make a decision on sensible thresholds? (both overall and % decrease)?

**Reply by helio-frota** [reply]
not yet, I think that will be part of ref-arch

**Comment by helio-frota** [comment]
I think we have done with this also.
in the case of travis and some repos with no CI, please take a look on other discussion thread https://github.com/nodeshift/new-module-starter/discussions/8
